### PR TITLE
fix: develop.sh: do not clobber existing login, pre-build coder binary

### PR DESCRIPTION
**Pull request by johnstcn**
I found that the `develop.sh` script can clobber your existing Coder login, so I made a `.coderv2` directory to hold the 'development' config directory, and added a shim to run an existing coder binary with that specific config dir.

I also figured it makes sense to build the Coder binary once instead of using `go run` a bunch of times in `develop.sh`, which makes it a bit faster.

Comments, suggestions, etc. welcome!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -40,3 +40,4 @@ site/out/
 
 .vscode/*.log
 **/*.swp
+.coderv2/*
```
Comment (jsjoeio) [inline .gitignore:43]: ❓ Should we call this `coderOSS`? That way it aligns with the name we use in the docs.

Comment (johnstcn) [reply .gitignore:43]: I just called it that because we already store stuff in `~/.config/coderv2` -- should we rename this config dir as well?

Comment (Kira-Pilot) [reply .gitignore:43]: I like `coderOSS`!

Comment (johnstcn) [reply .gitignore:43]: @jsjoeio is there an issue for this possible rename?

Comment (jsjoeio) [reply .gitignore:43]: Not that I know of - just something I thought of now. Would you like me to make one?

**Review by Emyrk — APPROVED** [review]

**Review by Emyrk — COMMENTED** [review]
Aside from the oss stuff, this is helpful

File: .gitignore (hunk 1 above)
Comment (johnstcn) [reply .gitignore:43]: Yes please, thanks!
It's probably eaiser to do sooner than later! Might be good to bring up in grooming next-next-week.

Comment (jsjoeio) [reply .gitignore:43]: Done! https://github.com/coder/coder/issues/2768

File: scripts/coder-dev.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+source "${SCRIPT_DIR}/lib.sh"
+PROJECT_ROOT=$(cd "$SCRIPT_DIR" && git rev-parse --show-toplevel)
+
+CODER_DEV_DIR="$PROJECT_ROOT/.coderv2/"
+CODER_DEV_BIN="${CODER_DEV_DIR}/coder"
+if [[ ! -d "${CODER_DEV_DIR}" ]]; then
+	mkdir -p "${CODER_DEV_DIR}"
+fi
+
+if [[ ! -x "${CODER_DEV_BIN}" ]]; then
```
Comment (mafredri) [inline scripts/coder-dev.sh:18]: Idea: Should we try to detect outdated binaries (e.g. long running `develop.sh` scripts?). Something like `if ! coder version | grep -q $(git rev-parse HEAD)`, then print warning?

File: scripts/develop.sh [hunk 3]
```diff
@@ -36,33 +42,33 @@ fi
 (
 	# If something goes wrong, just bail and tear everything down
 	# rather than leaving things in an inconsistent state.
-	trap 'kill -INT -$$' ERR
+	trap 'kill -TERM -$$' ERR
```
Comment (mafredri) [inline scripts/develop.sh:45]: I think this will prevent graceful shutdown of the server (e.g. resource cleanup), do we want to avoid that?

**Review by mafredri — APPROVED** [review]
Nice changes!

File: scripts/coder-dev.sh (hunk 2 above)
Comment (johnstcn) [reply scripts/coder-dev.sh:18]: That sounds fun!

File: scripts/develop.sh (hunk 3 above)
Comment (johnstcn) [reply scripts/develop.sh:45]: Should only trigger on an error and if we encounter an error then stuff hasn't even managed to start, so we should be fine.

Comment (mafredri) [reply scripts/develop.sh:45]: Oh right, that makes sense! Ignore me. 😄

**Pull request merged**